xdsl/printer.py

```diff
@@ -534,10 +535,25 @@ def print_attribute(self, attribute: Attribute) -> None:
 
         if isinstance(attribute, FloatAttr):
             value = attribute.value
-            attr_type = cast(
-                FloatAttr[Float16Type | Float32Type | Float64Type], attribute
-            ).type
-            self.print_string(f"{value.data:.6e} : ")
+            attr_type = cast(AnyFloatAttr, attribute).type
+            if np.isnan(value.data) or np.isinf(value.data):
+                if isinstance(attr_type, Float16Type):
+                    self.print_string(
+                        f"{hex(np.float16(value.data).view('uint16'))} : "
+                    )
+                elif isinstance(attr_type, Float32Type):
+                    self.print_string(
+                        f"{hex(np.float32(value.data).view('uint32'))} : "
```

Comment:
```suggestion
                        f"{hex(convert_f32_to_u32(inf))} : "
```

Reply:
Would need a few more functions

Reply:
I'm not sure what `convert_f32_to_u32` is or how it would be implemented?

Reply:
It's in xDSL already :) Could you please let the original commenter resolve the conversations, that way they don't accidentally get missed in subsequent reviews

Reply:
The point was to remove np?

Reply:
ctypes unfortunately does not support f16.

Reply:
Ah OK, thanks for looking into it